src/libsystemd/sd-device/sd-device.c

```diff
@@ -1873,13 +1872,33 @@ _public_ int sd_device_set_sysattr_value(sd_device *device, const char *sysattr,
                 return 0;
         }
 
+        value_len = strlen(_value);
+
+        /* drop trailing newlines */
+        while (value_len > 0 && _value[value_len - 1] == '\n')
+                _value[--value_len] = '\0';
```

Comment:
i really don't like that we clobber the input parameter here... This is against our CODING_STYLE really. We should copy this before we chop off newlines, not after